### PR TITLE
Issue2204 2264 cooling plant

### PR DESCRIPTION
**Pull request by JingWang-CUB**
This PR includes the generic cooling plant model and its example model for the URBANopt interface branch (issue #2264 ).

@khinkelman could you please perform a minimum technical review on this PR? For your information, I did use one `RealExpression` block in the plant model and I believe it's inevitable. Can we keep it just for now? Additionally, I haven't run the model using JModelica. If you could do that, it will be great! Thank you!

### CODE REVIEW

**Review by khinkelman — CHANGES_REQUESTED** [review]
Thanks @JingWang-CUB!
Your use of `realExpression` seems reasonable to me in these cases. I have no problem keeping them. 

I've completed the minimum technical review. Below is the summary. Please let me know if you have any comments/questions. Thanks!

**Minimum Technical Review Summary**
- [ ] Follow the LBNL Style Guide, User Guide Best Practices, and User Guide Conventions 
- [X] The models contain warnings in initial equation and information sections
- [ ] Branch passes CI testing (pending)
- [X] Verified that there are no unintended changes 
- [X] Block diagrams are used to compute variables as much as possible
- [X] Each instance has a description string
- [X] Each model has a `defaultComponentName`, except the base classes
- [X] A minimum documentation is provided for each class.
- [X] The main models and blocks have a corresponding validation or example model  
- [X] The main models have a simulation script with reference results stored in the library
- [ ] Simulations pass in JModelica
Please see errors in the html diagnostics I sent you, generated by running
`jm_ipython.sh jmodelica.py Buildings.Experimental.DHC.CentralPlants.Cooling.Examples.CoolingPlantOpenLoop`
- [X] modelica-fmt was run on the modified packages
- [X] Run unit tests

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo [hunk 1]
```diff
@@ -0,0 +1,460 @@
+within Buildings.Experimental.DHC.CentralPlants.Cooling;
+model Plant
+  "District cooling plant model"
+  package Medium=Buildings.Media.Water
+    "Medium model";
+  parameter Integer numChi(
+    min=1,
+    max=2)=2
+    "Number of chillers, maximum is 2";
+  parameter Boolean show_T=true
```
[the remark this answers is not in the document]
Comment (JingWang-CUB) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo:10]: Done

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+  replaceable parameter Buildings.Fluid.Chillers.Data.ElectricEIR.Generic perChi
+    "Performance data of chiller"
+    annotation (Dialog(group="Chiller"),choicesAllMatching=true,Placement(transformation(extent={{98,82},{112,96}})));
+  parameter Modelica.SIunits.MassFlowRate mCHW_flow_nominal
+    "Nominal chilled water mass flow rate"
+    annotation (Dialog(group="Chiller"));
+  parameter Modelica.SIunits.Pressure dpCHW_nominal
+    "Pressure difference at the chilled water side"
+    annotation (Dialog(group="Chiller"));
+  parameter Modelica.SIunits.Power QEva_nominal
```
[the remark this answers is not in the document]
Comment (JingWang-CUB) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo:23]: You're right, `HeatFlowRate` is a better unit here. I've changed it. 

`Eva` in `QEva_nominal` refers to the evaporation side of the chiller. I believe it is technically correct. Considering consistency with former PRs, let's leave it as is for now. We could change it later during the formal review for the master branch.

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+  Buildings.Fluid.Sensors.MassFlowRate senMasFloByp(
+    redeclare final package Medium=Medium)
+    "Chilled water bypass valve mass flow meter"
+    annotation (Placement(transformation(extent={{10,-10},{-10,10}},rotation=-90,origin={80,-30})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTCHWSup(
+    redeclare final package Medium=Medium,
+    final m_flow_nominal=mCHW_flow_nominal)
+    "Chilled water supply temperature"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}},rotation=0,origin={130,-50})));
+  Buildings.Experimental.DHC.CentralPlants.Cooling.Controls.ChilledWaterPumpSpeed CHWPumCon(
```
[the remark this answers is not in the document]
Comment (JingWang-CUB) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo:195]: Block name changed to `chiWatPumCon`.

**Comment by JingWang-CUB** [comment]
@khinkelman all issues have been addressed including the JModelica errors. Once the PR #2273 has been merged. This branch should be ready to be merged with `issue2204_gmt_mbl`.

**Comment by AntoineGautier** [comment]
@JingWang-CUB Ping me when you correct the remaining CI issues (https://github.com/lbl-srg/modelica-buildings/runs/1455499561) and I can directly pull the changes from your fork if you want. No need to open another PR then.

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo (hunk 2 above)
[the remark this answers is not in the document]
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo]: Sounds good to me. Thanks!

**Comment by khinkelman** [comment]
@JingWang-CUB : It looks like the `DHC.EnergyTransferStations.Cooling` models have now been deleted in this PR. Was this intentional? I believe this would cause these models to be deleted once merged with `issue2204_gmt_mbl` and should be corrected prior to merging.

**Comment by JingWang-CUB** [comment]
> @JingWang-CUB : It looks like the `DHC.EnergyTransferStations.Cooling` models have now been deleted in this PR. Was this intentional? I believe this would cause these models to be deleted once merged with `issue2204_gmt_mbl` and should be corrected prior to merging.

@khinkelman Thanks for noticing this. I did merge and revert merging the ETS models from `issue2204_gmt_mbl` on my fork. I did this because I realized the merging should not happen on the first PR but the second one. I will notify Antoine to pull from `issue2204_gmt_mbl` before he merges this PR.

**Comment by JingWang-CUB** [comment]
> @JingWang-CUB Ping me when you correct the remaining CI issues (https://github.com/lbl-srg/modelica-buildings/runs/1455499561) and I can directly pull the changes from your fork if you want. No need to open another PR then.

@AntoineGautier The CI errors have been fixed on my fork: https://github.com/JingWang-CUB/modelica-buildings/tree/issue2204_2264_cooling_plant. Please feel free to pull the commits. Thank you very much!

**Comment by AntoineGautier** [comment]
@JingWang-CUB Pulling `issue2204_gmt_mbl` does not reintroduce the deleted files because git detects that they were deleted at an ulterior timestamp so it considers the deletion as the up-to-date state of the branch, see:

```
(base) ~/gitrepo/modelica-buildings[issue2204_2264_cooling_plant]$ git pull origin issue2204_gmt_mbl
From https://github.com/lbl-srg/modelica-buildings
 * branch                  issue2204_gmt_mbl -> FETCH_HEAD
Already up to date.
(base) ~/gitrepo/modelica-buildings[issue2204_2264_cooling_plant]$ git diff origin/issue2204_gmt_mbl --name-status 
A       Buildings/Experimental/DHC/CentralPlants/Cooling/Examples/CoolingPlantOpenLoop.mo
M       Buildings/Experimental/DHC/CentralPlants/Cooling/Examples/package.order
A       Buildings/Experimental/DHC/CentralPlants/Cooling/Plant.mo
M       Buildings/Experimental/DHC/CentralPlants/Cooling/package.order
D       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/CoolingDirectControlledReturn.mo
D       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/CoolingDirectUncontrolled.mo
D       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/CoolingIndirect.mo
D       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingDirectControlledReturn.mo
D       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingDirectUncontrolled.mo
D       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingIndirectClosedBuildingLoop.mo
D       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingIndirectOpenLoops.mo
M       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/Examples/package.order
M       Buildings/Experimental/DHC/EnergyTransferStations/Cooling/package.order
D       Buildings/Resources/Images/Experimental/DHC/EnergyTransferStations/CoolingDirectControlledReturn.PNG
D       Buildings/Resources/Images/Experimental/DHC/EnergyTransferStations/CoolingDirectControlledReturn.svg
D       Buildings/Resources/Images/Experimental/DHC/EnergyTransferStations/CoolingDirectUncontrolled.PNG
D       Buildings/Resources/Images/Experimental/DHC/EnergyTransferStations/CoolingDirectUncontrolled.svg
D       Buildings/Resources/Images/Experimental/DHC/EnergyTransferStations/CoolingIndirect.png
D       Buildings/Resources/Images/Experimental/DHC/EnergyTransferStations/CoolingIndirect.svg
A       Buildings/Resources/ReferenceResults/Dymola/Buildings_Experimental_DHC_CentralPlants_Cooling_Examples_CoolingPlantOpenLoop.txt
D       Buildings/Resources/ReferenceResults/Dymola/Buildings_Experimental_DHC_EnergyTransferStations_Cooling_Examples_CoolingDirectControlledReturn.txt
D       Buildings/Resources/ReferenceResults/Dymola/Buildings_Experimental_DHC_EnergyTransferStations_Cooling_Examples_CoolingDirectUncontrolled.txt
D       Buildings/Resources/ReferenceResults/Dymola/Buildings_Experimental_DHC_EnergyTransferStations_Cooling_Examples_CoolingIndirectClosedBuildingLoop.txt
D       Buildings/Resources/ReferenceResults/Dymola/Buildings_Experimental_DHC_EnergyTransferStations_Cooling_Examples_CoolingIndirectOpenLoops.txt
A       Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Examples/CoolingPlantOpenLoop.mos
D       Buildings/Resources/Scripts/Dymola/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingDirectControlledReturn.mos
D       Buildings/Resources/Scripts/Dymola/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingDirectUncontrolled.mos
D       Buildings/Resources/Scripts/Dymola/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingIndirectClosedBuildingLoop.mos
D       Buildings/Resources/Scripts/Dymola/Experimental/DHC/EnergyTransferStations/Cooling/Examples/CoolingIndirectOpenLoops.mos
A       Buildings/Resources/Scripts/OpenModelica/compareVars/Buildings.Experimental.DHC.CentralPlants.Cooling.Examples.CoolingPlantOpenLoop.mos
D       Buildings/Resources/Scripts/OpenModelica/compareVars/Buildings.Experimental.DHC.EnergyTransferStations.Cooling.Examples.CoolingDirectControlledReturn.mos
D       Buildings/Resources/Scripts/OpenModelica/compareVars/Buildings.Experimental.DHC.EnergyTransferStations.Cooling.Examples.CoolingDirectUncontrolled.mos
D       Buildings/Resources/Scripts/OpenModelica/compareVars/Buildings.Experimental.DHC.EnergyTransferStations.Cooling.Examples.CoolingIndirectClosedBuildingLoop.mos
D       Buildings/Resources/Scripts/OpenModelica/compareVars/Buildings.Experimental.DHC.EnergyTransferStations.Cooling.Examples.CoolingIndirectOpenLoops.mos
```

You need to manually check out those files from `issue2204_gmt_mbl` (or restart from `issue2204_gmt_mbl` and reintroduce the files you modified). Please let me know when you do that and I will pull your branch.

**Comment by JingWang-CUB** [comment]
@AntoineGautier I have manually reintroduced the ETS models on my fork here: https://github.com/JingWang-CUB/modelica-buildings/tree/issue2204_2264_cooling_plant. Sorry for the trouble caused by my false actions on the timeline. Let me know if this worked fine or not. Thank you!

**Comment by AntoineGautier** [comment]
@JingWang-CUB CI fails because you have experiment attributes in some of your example or validation models that must be removed. Could you please remove them, and also specify `cvode` as the default solver in all `*.mos` scripts (this is now the default for all models in MBL). Please let me know when you do that and I will pull your branch directly.

**Comment by JingWang-CUB** [comment]
@AntoineGautier It's done on my fork. Thank you for pulling it!

**Pull request merged**